app/client-app/src/app/pages/Editor/components/PageList/PageHeader.js

```diff
@@ -1,12 +1,33 @@
-import { Flex, useColorMode, Icon } from "@chakra-ui/react";
+import { Flex, useColorMode, Icon, Input } from "@chakra-ui/react";
+import { useState, useEffect, useRef } from "react";
 import LightHeading from "components/core/LightHeading";
 import { FaGripVertical } from "react-icons/fa";
 import PageActionButtons from "./PageActionButtons";
 import { defaultColorMode } from "themes";
-
+import axios from "axios";
+import { useFetchSurvey } from "app/contexts/FetchSurvey";
+import { authorization_BearerToken, withHeaders, contentType_AppJson } from "api/helpers";
 const PageHeader = ({ page, order, dragHandleProps }) => {
   const { colorMode } = useColorMode();
   const headerStyle = { light: { bg: "gray.300" }, dark: { bg: "gray.600" } };
+  const { id, mutate } = useFetchSurvey();
+  const [name, setName] = useState(page.name);
+  const firstUpdate = useRef(true);
+  useEffect(() => {
+    if (firstUpdate.current == true) {
+      firstUpdate.current = false
+      return
+    }
+    const delayDebounceFn = setTimeout(async () => {
+      axios.put(
+        `/api/surveys/${id}/pages/${page.id}/name`,
+        JSON.stringify(name),
+        withHeaders(contentType_AppJson, await authorization_BearerToken())
+      ).then(res => console.log(res))
```

Comment:
don't use old Promise (`.then()` etc.) syntax inside `async` functions - use `await` instead. [`async/await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) is the preferred style anywhere it can be used (almost everywhere in ES2022)

Reply:
Also don't leave `console.log` anywhere in code going into `main`.